src/WinGet.RestSource/Models/Schemas/ManifestSearchResponse.cs

```diff
@@ -260,9 +260,7 @@ public static List<ManifestSearchResponse> Consolidate(List<ManifestSearchRespon
             {
                 // Create Consolidation Predicate for search Response..

```

Comment:
This feels overly complicated to me, why not:
`
                // If exists in results update otherwise add
                ManifestSearchResponse match = list.FirstOrDefault(x => x.PackageIdentifier == response.PackageIdentifier);
                if (match != null)
                {
                    // Verify versions exists
                    if (match.Versions == null && response.Versions != null)
                    {
                        match.Versions = new SearchVersions();
                    }

                    match.Versions.Merge(response.Versions);
                }
                else ...
`